File: frontend/src/types/externalApproval.ts
export type ExternalApprovalType = "bb.plugin.app.feishu";

export type ExternalApprovalEvent = {
  type: ExternalApprovalType;
  action: ExternalApprovalEventActionType;
  stageName: string;
};

export type ExternalApprovalEventActionType = "APPROVE" | "REJECT";

File: backend/legacyapi/external_approval.go
package api

// ExternalApprovalType is the type of the ExternalApproval.
type ExternalApprovalType string

const (
	// ExternalApprovalTypeFeishu is the ExternalApproval from feishu.
	ExternalApprovalTypeFeishu ExternalApprovalType = "bb.plugin.app.feishu"
	// ExternalApprovalTypeRelay is the ExternalApproval from relay.
	ExternalApprovalTypeRelay ExternalApprovalType = "bb.plugin.app.relay"
)
